lib/tooltip.js

```diff
@@ -0,0 +1,439 @@
+import React, { useState, useEffect } from 'react';
+import styled, { css } from 'styled-components';
+import PropTypes from 'prop-types';
+import { code, CodeExample, PropsDefinition, Prop } from './story-utils';
+import { useUniqueId } from './hooks/use-unique-id';
+
+const TYPES = ['action', 'info'];
+const ALIGNMENTS = ['left', 'right', 'center', 'top'];
+
+const TooltipContainerEl = styled.div`
+  position: relative;
+  box-sizing: border-box;
+  display: inline-block;
+`;
+
+const leftAndLefBefore = css`
```

Comment:
this is what you were blogging about, right?

Reply:
Yep. The css`` function isn't strictly necessary here, but would enable these re-usable things to also consume props. In my use case where I don't need props, they make VS Code's styled components syntax highlighter plugin work on these lines. (Otherwise they'd all be the simple string color)